backends/platform/iphone/iphone_video.mm

```diff
@@ -158,31 +166,74 @@ - (void)createContext {
 	}
 }
 
+- (void)setupGestureRecognizers {
+	UISwipeGestureRecognizer *swipeRight = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(twoFingersSwipeRight:)];
+	swipeRight.direction = UISwipeGestureRecognizerDirectionRight;
+	swipeRight.numberOfTouchesRequired = 2;
+
+	UISwipeGestureRecognizer *swipeLeft = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(twoFingersSwipeLeft:)];
+	swipeLeft.direction = UISwipeGestureRecognizerDirectionLeft;
+	swipeLeft.numberOfTouchesRequired = 2;
+
+	UISwipeGestureRecognizer *swipeUp = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(twoFingersSwipeUp:)];
+	swipeUp.direction = UISwipeGestureRecognizerDirectionUp;
+	swipeUp.numberOfTouchesRequired = 2;
+
+	UISwipeGestureRecognizer *swipeDown = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(twoFingersSwipeDown:)];
+	swipeDown.direction = UISwipeGestureRecognizerDirectionDown;
+	swipeDown.numberOfTouchesRequired = 2;
+
+	UITapGestureRecognizer *doubleTapTwoFingers = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(twoFingersDoubleTap:)];
+	doubleTapTwoFingers.numberOfTapsRequired = 2;
+	doubleTapTwoFingers.numberOfTouchesRequired = 2;
+
+	[self addGestureRecognizer:swipeRight];
+	[self addGestureRecognizer:swipeLeft];
+	[self addGestureRecognizer:swipeUp];
+	[self addGestureRecognizer:swipeDown];
+	[self addGestureRecognizer:doubleTapTwoFingers];
+}
+
+- (CGFloat)optimalScale {
+	CGFloat screenScale = [[UIScreen mainScreen] scale];
+	if (screenScale < 2) return screenScale;
+
+	if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
+		return 1;
+	}
+
+	CGSize screenSize;
+	UIScreen *mainScreen = [UIScreen mainScreen];
+	if ([mainScreen respondsToSelector:@selector(nativeBounds)]) {
+		screenSize = [mainScreen nativeBounds].size;
```

Comment:
```
backends/platform/iphone/iphone_video.mm:208:28: warning: instance method '-nativeBounds' not found (return type defaults to 'id') [-Wobjc-method-access]
                screenSize = [mainScreen nativeBounds].size;                                                                
                                         ^~~~~~~~~~~~                                                                       
backends/platform/iphone/iphone_video.mm:208:42: error: property 'size' not found on object of type 'id'         
                screenSize = [mainScreen nativeBounds].size;
                                                       ^
```

This time you check whether the selector is actually available. But it doesn't know about the proper type here. Not sure how to resolve this, this is a bit different from the code we used for the scale selector handling...